generator/restriction_dumper.cpp

```diff
@@ -0,0 +1,103 @@
+#include "generator/restriction_dumper.hpp"
+#include "generator/intermediate_elements.hpp"
+#include "generator/osm_id.hpp"
+#include "generator/restriction_collector.hpp"
+
+#include "indexer/routing.hpp"
+
+#include "base/logging.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/fstream.hpp"
+#include "std/string.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace
+{
+using namespace routing;
+
+vector<string> const kRestrictionTypesNo = {"no_right_turn",  "no_left_turn", "no_u_turn",
+                                            "no_straight_on", "no_entry",     "no_exit"};
+vector<string> const kRestrictionTypesOnly = {"only_right_turn", "only_left_turn",
+                                              "only_straight_on"};
+
+/// \brief Converts restriction type form string to RestrictionCollector::Type.
+/// \returns Fisrt item is a result of conversion. Second item is true
+/// if convertion was successful and false otherwise.
+pair<Restriction::Type, bool> TagToType(string const & type)
+{
+  if (find(kRestrictionTypesNo.cbegin(), kRestrictionTypesNo.cend(), type) !=
+      kRestrictionTypesNo.cend())
+  {
+    return make_pair(Restriction::Type::No, true);
```

Comment:
Yes. I'm changing to "bool TagToType(string const & tag, Restriction::Type & type)"